x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/SchemaUtil.java

```diff
@@ -68,10 +70,18 @@ public static void deduceMappings(
         Map<String, String> aggregationTypes = new HashMap<>();
         // collects the fieldnames and target fieldnames used for grouping
         Map<String, String> fieldNamesForGrouping = new HashMap<>();
+        // collects the target mapping types used for grouping
+        Map<String, String> fieldTypesForGrouping = new HashMap<>();
 
         config.getGroupConfig()
             .getGroups()
-            .forEach((destinationFieldName, group) -> { fieldNamesForGrouping.put(destinationFieldName, group.getField()); });
+            .forEach((destinationFieldName, group) -> {
+                fieldNamesForGrouping.put(destinationFieldName, group.getField());
+                if (group.getMappingType() != null) {
```

Comment:
nit:

```
if (group.getMappingType() != null) {
    fieldTypesForGrouping.put(destinationFieldName, group.getMappingType());
} else {
    fieldNamesForGrouping.put(destinationFieldName, group.getField());
}

```

Reply:
This won't work, `fieldNamesForGrouping` always needs the group config field name so that it can add it to the mapping. Sometimes the group config has a desired type and that will get added to the new `fieldTypesForGrouping` map. 

Adding comments explaining such.

Reply:
my bad, human parsing error: field**Types**ForGrouping vs. field**Names**ForGrouping - when reading over it the first time, it looked the same.